src/content_main.js

```diff
@@ -171,42 +162,134 @@ async function storeRecentTimeSpent(recentTimeSpent) {
+ */
+function trackPaymentPointer() {
+	let cachedPaymentPointer = null;
+
+	setInterval(async () => {
+		const paymentPointerOnPage = getPaymentPointerFromPage();
+
+		const isNewPaymentPointer =
+			(paymentPointerOnPage !== null) &&
+			(paymentPointerOnPage !== cachedPaymentPointer);
+		if (isNewPaymentPointer) {
+			// Note that paymentPointerOnPage may be newly added to the current page and tab,
+			// but it is possible that this payment pointer was recently validated in another tab.
+			// Therefore it is necessary to use handleValidationAndSetExtensionIcon which checks the
+			// browser storage to see if the payment pointer has been recently validated.
+			handleValidationAndSetExtensionIcon(paymentPointerOnPage);
+			cachedPaymentPointer = paymentPointerOnPage;
+		}
+	}, NEW_PAYMENT_POINTER_CHECK_RATE_MS);
+}
 
-	if (null === monetizationMeta) {
-		/* No monetization meta tag provided */
+/**
+ * handleUpdatingExtensionIconOnTabVisibilityChange ensures that the extension icon is updated
+ * to the correct state when switching between tabs.
+ */
+function handleUpdatingExtensionIconOnTabVisibilityChange() {
+	document.addEventListener('visibilitychange', (event) => {
+		const tabIsVisible = !document.hidden;
+		if (tabIsVisible) {
+			const paymentPointerOnPage = getPaymentPointerFromPage();
+			if (paymentPointerOnPage !== null) {
+				handleValidationAndSetExtensionIcon(paymentPointerOnPage);
+			}
+		}
+	});
+}
+
+/**
+ * Takes a payment pointer and validates it if it has not been validated recently. If the payment
+ * pointer is sucessfuly validated then the extension icon is set to its monetized state, otherwise
+ * it is set to its unmonetized state.
+ *
+ * @param {string} paymentPointer the payment pointer to validate. Extension icon is set to its
+ * monetized image if the validation is successful.
+ * @returns {Promise<>} a promise that resolves when validation is finished.
+ */
+async function handleValidationAndSetExtensionIcon(paymentPointer) {
+	if (await isPaymentPointerRecentlyValidated(paymentPointer)) {
+		setExtensionIconMonetizationState(true);
 	} else {
 		if (await isPaymentPointerValid(paymentPointer)) {
-			isValid = true;
+			// isPaymentPointerValid validates the payment pointer, so we store the payment
+			// pointer and the UTC timestamp for the time it was validated at (now).
+			await storeDataIntoAkitaFormat({
+				paymentPointer: paymentPointer,
+				validationTimestamp: Date.now()
+			}, AKITA_DATA_TYPE.PAYMENT);
+			setExtensionIconMonetizationState(true);
+		} else {
+			setExtensionIconMonetizationState(false);
 		}
 	}
+}
+
+/**
+ * Checks the browser storage to see if Akita has validated the given payment pointer and stored a
+ * validation timestamp for it.
+ * Returns true if there is a timestamp retrieved from storage and the
+ * timestamp is recent (less than PAYMENT_POINTER_VALIDATION_RATE_MS from present time).
+ * Returns false otherwise.
```

Comment:
This should be covered in the `@returns` section and does not need to be duplicated